sale_rental/wizard/create_rental_product.py

```diff
@@ -0,0 +1,93 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Sale Rental module for Odoo
+#    Copyright (C) 2014-2015 Akretion (http://www.akretion.com)
+#    @author Alexis de Lattre <alexis.delattre@akretion.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, _
+import openerp.addons.decimal_precision as dp
+
+
+class CreateRentalProduct(models.TransientModel):
+    _name = 'create.rental.product'
+    _description = 'Create the Rental Service Product'
+
+    @api.model
+    def _default_name(self):
+        assert self.env.context.get('active_model') == 'product.product',\
+            'Wrong underlying model, should be product.product'
+        hw_product = self.env['product.product'].browse(
+            self.env.context['active_id'])
+        return _('Rental of a %s') % hw_product.name
+
+    @api.model
+    def _default_code(self):
+        assert self.env.context.get('active_model') == 'product.product',\
+            'Wrong underlying model, should be product.product'
+        hw_product = self.env['product.product'].browse(
+            self.env.context['active_id'])
+        return _('RENT-%s') % hw_product.default_code
```

Comment:
The stockable product that is used as a reference to create the service rental could have the default code set as empty. In that case the current code sets "RENT-false". I suggest to add the default code for the created rental service only if the associated product has a default_code set.